packages/babel-parser/test/fixtures/experimental/record-and-tuple/disambiguate-regex-hash/input.js

```diff
@@ -0,0 +1 @@
+(#{})/foo
```

Comment:
This case fails on current `main`, too. Because a token context is not pushed and `tt.braceR.updateContext` incorrectly set `this.state.exprAllowed = true` (when `context.length` is 1) so `/foo` is parsed as unterminated regex.